Remove ert-storage as dependency

dafeda opened the pull request:
## Pre review checklist

- [ ] Read through the code changes carefully after finishing work
- [ ] Make sure tests pass locally (after every commit!)
- [ ] Prepare changes in small commits for more convenient review (optional)
- [ ] PR title captures the intent of the changes, and is fitting for release notes.
- [ ] Updated documentation
- [ ] Ensured that unit tests are added for all new behavior (See 
    [Ground Rules](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md#ground-rules)),
    and changes to existing code have good test coverage.

## Pre merge checklist
- [x] Added appropriate release note label
- [ ] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).

<!--
Adding labels helps the maintainers when writing release notes. This is the
[list of release note
labels](https://github.com/equinor/ert/labels?q=release-notes).
-->

frode-aarstad commented:
Should we also use the opportunity to rename `dark_storage` ?

frode-aarstad commented on src/ert/services/_storage_main.py:
```diff
@@ -121,20 +121,14 @@ def run_server(args: Optional[argparse.Namespace] = None, debug: bool = False) -
+    # Dark Storage imports from ERT Storage, which connects to the database
+    # at startup. We set the database URL to an SQLite in-memory database so
+    # that the import succeeds.
+    os.environ["ERT_STORAGE_DATABASE_URL"] = "sqlite://"
```
Think we can remove the DB references. There are also some references to `sqlite` around the code

pinkwah reviewed (commented):
Please also put a link to the github repo and the commit from which you took this code, so that the git history is at least somewhat preserved.

pinkwah commented:
> Should we also use the opportunity to rename `dark_storage` ?

I would like to at least add `_server` as postfix. I personally think `_storage_server` would be most descriptive of what this is supposed to be. Or `_onprem_storage_server`, to distinguish this from a possible future separate service that we would like to switch to.

dafeda commented:
> Should we also use the opportunity to rename `dark_storage` ?

I'd rather we do that in a separate PR I think.

dafeda commented:
> Please also put a link to the github repo and the commit from which you took this code, so that the git history is at least somewhat preserved.

I've squashed and added a link to the latest commit from ert-storage in the commit message body.
